client/layout/guided-tours/index.js

```diff
@@ -23,8 +23,9 @@ class GuidedTours extends Component {
 		return this.props.tourState !== nextProps.tourState;
 	}
 
-	start = ( { tour, tourVersion: tour_version } ) => {
+	start = ( { tour, tourVersion: tour_version, step } ) => {
 		if ( tour && tour_version ) {
+			this.props.nextGuidedTourStep( { tour, step } );
```

Comment:
Because the selector's decision (null or a string like `main`) is the only requirement for a tour to start. So the sequence in terms of data and actions is:
0. No tour is active
1. Some action is performed that gets logged in `ui.actionLog`
2. That state change triggers a new evaluation of the GT selectors with the new state, ordered by the `connect`-ed `layout/guided-tours` component
3. The selectors determine that the recent action is a trigger for tour `main`, so `tour="main"` is passed to the GT components. `step` defaults to `'init'` anyway
4. The GT `main` intro step is rendered, and we haven't needed to add `GUIDED_TOUR_*` actions to `actionLog`
5. The user presses Next or Quit, and those do dispatch `GUIDED_TOUR_UPDATE` actions, and their payload explicitly states the current tour, `main`, and the requested next step, e.g. `second`

With this PR, the main difference is on step 4, where the component will dispatch an initial GT actions so that, from then on, the selectors actually _know_ [^1] that there is an active tour, and so they won't try to find a potentially different eligible tour until the user ends the ongoing one.

[^1]: this talk about _selectors that know things_ makes me think of _the sentinels_ in The Matrix.